### ISSUE TITLE
Session doesn't perform retries

### ISSUE THREAD

**Issue by isaacnorman82** [jasonrollins/shareplum#83]
You could easily set up some retries on the request session to cope with flaky connections:

from urllib3 import Retry

self._session = requests.Session()
retry = Retry(total=5, read=5, connect=5, backoff_factor=0.3, status_forcelist=[500, 502, 503, 504])
self._session.mount('https://', HTTPAdapter(max_retries=retry))

**Comment by jasonrollins** [comment]
If you do the session.mount('http://') part does it overwrite the verify_ssl functionality in session.post()?

**Comment by isaacnorman82** [comment]
I think I've got code that does both, I'll have a look tomorrow and paste an example.

**Comment by isaacnorman82** [comment]
I see you're using something called SSLAdapter from request_toolbelt:

```
if ssl_version is not None:
            self._session.mount("https://", SSLAdapter(ssl_version))
```

Looking at the requets_toolbelt repo I see SSLAdaptor extends HTTPAdaptor and passes through arguments to the constructor, so you would be able to just do something like this:

```
retry = Retry(total=5, read=5, connect=5, backoff_factor=0.3, status_forcelist=[500, 502, 503, 504])
http_adaptor = HTTPAdatpter(max_retries=retry)
https_adaptor = http_adaptor
if ssl_version is not None:
            https_adaptor = SSLAdapter(ssl_version, max_retries=retry))

self._session.mount("https://", https_adaptor)
self._session.mount("http://", http_adaptor)
```

I'm connecting to <my tennant>.sharepoint.com and shareplum is curretnly using a 3 second timeout and it's being hit sometimes.  This would increase the timeout to 5s and add retries.

**Comment by jasonrollins** [comment]
Thanks!  I'm good with that.  I'll implement this over the weekend.

**Comment by jasonrollins** [comment]
This is not implemented in 0.5.0.  Thanks!
